### DISCUSSION TITLE
replace docx2txt?

### DISCUSSION THREAD

**Discussion by mophilly** [enoch3712/ExtractThinker#171] [General]
homebrew Python 3.12, MacOS 15.2. 
For a reason(s) I have yet to determine, the package docx2txt will not successfully install. It was working until about a week ago. I have read a great deal, asked for help in other forums and deleted/rebuilt the venv several times. Other packages, like pytesseract, install as expected. At this point I am frustrated and my real project is stuck. 

Given that docx2txt is years past its last update and is no longer maintained, it may have become incompatible in some fashion. E.g. it uses distutils. I wish I had reasonable detail for this problem.

What is involved to replace it with a modern functional equivalent?

**Comment by enoch3712** [comment]
Hey @mophilly!

Yes, you are 100% right, im doing that now. For you to have a context, comes from langchain adaptation.

https://python.langchain.com/docs/integrations/document_loaders/microsoft_word/

Now, on 0.0.33v you have Docx fully supported on markitdown.

https://enoch3712.github.io/ExtractThinker/core-concepts/document-loaders/markitdown/

So, you cannot run it correct? Even when not in use? Im switching to lazy loading of dependencies, **if you need this now**, i push the lazy loading for the docx2txt. **Takes me 2 mins no worries**

**Reply by mophilly** [reply]
I haven't tried updating to 0.0.33 as I have been focused on finagling docx2txt. 
I will read the links before I post further.

**Reply by mophilly** [reply]
You asked "you cannot run it". Right, I am stuck as the import docx2txt stymies the interpreter. 
the MarkItDown is a cool bit; I read a little about it over the weekend.

**Reply by enoch3712** [reply]
Ok, im fix that soon, no worries

**Reply by enoch3712** [reply]
https://github.com/enoch3712/ExtractThinker/issues/136

**Comment by enoch3712** [comment]
@mophilly 

https://github.com/enoch3712/ExtractThinker/releases/tag/v0.0.34

Its done! Now you have to install each documentloader depency, since now is much slimmer

**Comment by mophilly** [comment]
That is great! I have updated docx2txt to use setuptools, instead of the distutils. If I set its version to 0.9 it installs and works fine. The snag was the requirement in ExtractThinker, so #173  and #136 will help a lot. 

I am not a maintainer on the docx2txt project but I will reach out to the author; perhaps it can be rev’d.

**Reply by enoch3712** [reply]
I think i will remove it, eventually, as the market is going, you have MarkitDown and Docling, that will pretty much be the 90% of the markshare
